UPGRADE.md

```diff
@@ -1,10 +1,36 @@
 # Upgrade Guide
 
-_After `7.4.0` all `MINOR` and `MAJOR` version bumps will have upgrade notes 
-posted here._
+_`MAJOR` version bumps will have upgrade notes posted here._
+
+[2020-09-28] 7.x.x to 8.x.x
+-----------------------------
+
+### Overview
+
+Version `8.x.x` is the first version that officially drops support for Java 7. 
+
+- Removal of dependencies offering functionality included in Java 8 and beyond:
+    - [Guava concurrency, hashing, and charstreams](https://github.com/twilio/twilio-java/pull/575)
+        - `com.google.common.hash.Hash*` -> `org.apache.commons.codec.digest.DigestUtils`
+        - `com.google.common.io.CharStreams` -> `org.apache.commons.io.IOUtils`
+        - `com.google.common.util.concurrent.*` -> `java.util.concurrent`
+    - [Guava `com.google.common.collect.Range` removal]()
+        - Example: `setMessageDate(final Range<LocalDate> rangeMessageDate)` -> `setMessageDate(final LocalDate messageDateBefore, final LocalDate messageDateAfter)`
```

Comment:
A range could be bound by upper, lower, or both. Think this should be broken out to 2 methods similar to what's done in other languages: `setMessageDateBefore` and `setMessageDateAfter`. Also need an example of how to change my code. What should my code look like before and after?